src/tsolvers/THandler.cc

```diff
@@ -521,10 +521,11 @@ void THandler::declareAtom(PTRef tr) {
 }
 
 PTRef THandler::getSubstitution(PTRef tr) const {
-    auto const & subst = getSolverHandler().substs;
-    if (subst.has(tr)){
-        PtAsgn subs = subst[tr];
-        return subs.sgn == l_True ? subs.tr : PTRef_Undef;
+    auto const & substs = getSolverHandler().substs;
+    PTRef target;
+    if (substs.peek(tr, target)) {
+        assert(not substs.has(target)); // Must be closed
```

Comment:
That's the question, must it really be closed?

Reply:
Postponed to the next PR